x-pack/plugins/ml/public/application/jobs/jobs_list/components/job_details/job_messages_pane.tsx

```diff
@@ -46,18 +55,87 @@ export const JobMessagesPane: FC<JobMessagesPaneProps> = React.memo(
 
     const refreshMessage = useCallback(fetchMessages, [jobId]);
 
+    // Clear messages for last 24hrs and refresh jobs list
+    const clearMessages = useCallback(async () => {
+      setIsClearing(true);
+      try {
+        await clearJobAuditMessages(jobId);
+        setIsClearing(false);
+        if (typeof refreshJobList === 'function') {
+          refreshJobList();
+        }
+      } catch (e) {
+        setIsClearing(false);
+        toastNotificationService.displayErrorToast(
+          e,
+          i18n.translate('xpack.ml.jobMessages.clearJobAuditMessagesErrorTitle', {
+            defaultMessage: 'Error clearing job message warnings and errors',
+          })
+        );
+      }
+    }, [jobId]);
+
     useEffect(() => {
       fetchMessages();
     }, []);
 
+    const disabled = messages.length > 0 && messages[0].clearable === false;
+
+    const clearButton = (
+      <EuiButton
+        size="s"
+        isLoading={isClearing}
+        isDisabled={disabled}
+        onClick={clearMessages}
+        data-test-subj="mlJobMessagesClearButton"
+      >
+        <FormattedMessage
+          id="xpack.ml.jobMessages.clearMessagesLabel"
+          defaultMessage="Clear notifications"
+        />
+      </EuiButton>
+    );
+
     return (
-      <JobMessages
-        refreshMessage={refreshMessage}
-        messages={messages}
-        loading={isLoading}
-        error={errorMessage}
-        actionHandler={actionHandler}
-      />
+      <>
+        <EuiSpacer />
+        <EuiFlexGroup direction="column">
```

Comment:
Does it make sense to show the clear button if there are no `error` or `warning` notifications? For jobs with only `info` icons, the action will do nothing.

![image](https://user-images.githubusercontent.com/7405507/123838470-f09ecb80-d903-11eb-8e1f-d13c3401a1b8.png)

Reply:
Button is no longer shown unless there's a relevant 'auditMessage' attached to the job in the jobs list that causes the icon to show up. 9538177239a8b6b8f5b5b6ae1d9bf579d223fe41